### ISSUE TITLE
ProfileSpecToXmlPersistenceTest.java: testSaveEmptyProfileSpecAsXml Test fails

### ISSUE THREAD

**Issue by ross-spencer** [digital-preservation/droid#79]
In the following test I'm receiving a contradictory XML string to the expected empty PROFILE string. The string I receive is:

```
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Profile Id="untitled-1">
    <CreatedDate>2009-01-01T00:00:00+13:00</CreatedDate>
    <ProfileSpec>
        <Resources/>
    </ProfileSpec>
    <State>STOPPED</State>
    <SignatureFileVersion>26</SignatureFileVersion>
    <Throttle>0</Throttle>
</Profile>
```

As such, I wonder if the control test string should be modified from:

```
    String control = "<Profile Id=\"untitled-1\">"
            + "  <CreatedDate>2009-01-01T00:00:00Z</CreatedDate>"
            + "  <State>STOPPED</State>"
            + "  <Throttle>0</Throttle>"
            + "  <SignatureFileVersion>26</SignatureFileVersion>"
            + "  <ProfileSpec>"
            + "    <Resources/>" 
            + "  </ProfileSpec>" 
            + "</Profile>";
```

To:

```
     String control = "<Profile Id=\"untitled-1\">"
     + "    <CreatedDate>2009-01-01T00:00:00+13:00</CreatedDate>"
     + "    <ProfileSpec>"
     + "        <Resources/>"
     + "    </ProfileSpec>"
     + "    <State>STOPPED</State>"
     + "    <SignatureFileVersion>26</SignatureFileVersion>"
     + "    <Throttle>0</Throttle>"
     + "</Profile>";
```

This removes the ProfileSpec elements and Resource elements. 

It is not clear to me how this test is passing for others e.g. TNA at present and so perhaps additional documentation will be useful too if the test stands currently as correct. 

Tested on:

```
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.3 LTS
Release:    14.04
Codename:   trusty
```

There may be a cross-OS compatibility issue.

**Comment by timyip3** [comment]
Would you mind to paste your exception log?

**Comment by ross-spencer** [comment]
Hi timyip3 - it's a unit test, so no exception log. It's just a unit test failure. If you look at the test, it's pretty simple and so you if you really want to recreate it using the observed XML above you can create a control two variable to compare the test against so it will always compare against that string. You can do the following:

    XMLAssert.assertXMLEqual(new StringReader(control), controlTwo)

 You'll be able to see how it fails that way. 

Unit test extracted here: 

        @Test
          public void testSaveEmptyProfileSpecAsXml() throws Exception {

              SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");

              File file = new File("profiles/untitled-1/profile.xml");
              FileUtils.deleteQuietly(file);

              ProfileSpec profileSpec = new ProfileSpec();

              ProfileInstance profile = new ProfileInstance(ProfileState.INITIALISING);
              profile.changeState(ProfileState.STOPPED);
              profile.setUuid("untitled-1");
              profile.setProfileSpec(profileSpec);
              profile.setDateCreated(df.parse("2009-01-01 12:00:00"));
              profile.setSignatureFileVersion(26);

              profileSpecJaxbDao.saveProfile(profile, new File("profiles/untitled-1"));

              String control = "<Profile Id=\"untitled-1\">"
                      + "  <CreatedDate>2009-01-01T00:00:00Z</CreatedDate>"
                      + "  <State>STOPPED</State>"
                      + "  <Throttle>0</Throttle>"
                      + "  <SignatureFileVersion>26</SignatureFileVersion>"
                      + "  <ProfileSpec>"
                      + "    <Resources/>" 
                      + "  </ProfileSpec>" 
                      + "</Profile>";

              Reader test = new FileReader(file);
              XMLAssert.assertXMLEqual(new StringReader(control), test);

          }

**Comment by rhubner** [comment]
Hi,
fixed in #163 2d96cbe

Radek
